core/types/tx_pool.go

```diff
@@ -0,0 +1,102 @@
+package types
+
+import (
+	"io"
+	"math/big"
+
+	"github.com/pkg/errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+	staking "github.com/harmony-one/harmony/staking/types"
+)
+
+var (
+	// ErrUnknownPoolTxType is returned when attempting to assert a PoolTransaction to its concrete type
+	ErrUnknownPoolTxType = errors.New("unknown transaction type in tx-pool")
+)
+
+// PoolTransaction is the general transaction interface used by the tx pool
+type PoolTransaction interface {
```

Comment:
do you need to document each field in this interface?

Reply:
Actually comments are good for external parties that do not understanding entirely the code.

Reply:
I believe we force all exposed components to have documentation (via golint in travis check?)